qubes/app.py

```diff
@@ -674,6 +674,28 @@ def _setter_default_netvm(app, prop, value):
     return value
 
 
+def validate_kernel(obj, property_name: str, kernel: str) -> None:
+    """Helper function to validate existence of specific kernel"""
+    if not kernel:
+        return
+    dirname = os.path.join(
+        qubes.config.qubes_base_dir,
+        qubes.config.system_path['qubes_kernels_base_dir'],
+        kernel)
+    if not os.path.exists(dirname):
+        raise qubes.exc.QubesPropertyValueError(
+            obj, property_name, kernel,
```

Comment:
the second argument is supposed to be QubesProperty object, not str. `obj.property_get_def(property_name)` should do the trick

Reply:
> the second argument is supposed to be QubesProperty object, not str. `obj.property_get_def(property_name)` should do the trick

It is done. Unittests re-run succeeded.